tests/ptable/test_ptable_matplotlib.py

```diff
@@ -26,6 +26,8 @@
 
     from pymatgen.core import Composition
 
+df_ptable = df_ptable.copy()  # avoid changing the df in place
```

Comment:
Don't know if there will be a better solution (`pandas` doesn't seem to provide any method to make a dataframe immutable)?

Also this is pretty interesting (my first time to notice one unit test alters the behavior of another), in the case `test_ptable_matplotlib.py` seems to alter `df_table` and therefore `test_process_data.py` failed (just guessing).